src/sinks/azure_common/config.rs

```diff
@@ -96,22 +92,23 @@ pub fn build_healthcheck(
     client: Arc<ContainerClient>,
 ) -> crate::Result<Healthcheck> {
     let healthcheck = async move {
-        let request = client.get_properties().execute().await;
+        let response = client.get_properties().into_future().await;
 
-        match request {
+        let resp: crate::Result<()> = match response {
             Ok(_) => Ok(()),
             Err(reason) => Err(match reason.downcast_ref::<HttpError>() {
```

Comment:
Thanks for the input. I tried to change the code but run into a problem I don't know how to solve:
```rust
pub fn build_healthcheck(
    container_name: String,
    client: Arc<ContainerClient>,
) -> crate::Result<Healthcheck> {
    let healthcheck = async move {
        let response = client.get_properties().into_future().await;

        let resp: crate::Result<()> = match response {
            Ok(_) => Ok(()),
            Err(reason) => match reason.kind() {
                ErrorKind::HttpResponse {
                    status: azure_core::StatusCode::Forbidden,
                    ..
                } => Err(Box::new(HealthcheckError::InvalidCredentials)),
                ErrorKind::HttpResponse {
                    status: azure_core::StatusCode::NotFound,
                    ..
                } => Err(Box::new(HealthcheckError::UnknownContainer {
                    container: container_name,
                })),
                ErrorKind::HttpResponse {
                    status,
                    ..
                } => {
                    let result = StatusCode::from_u16(status.into());
                    match result {
                        Ok(status) => Err(Box::new(HealthcheckError::Unknown { status: status })),
                        Err(status) => Err("Unknown status code".into()),
                    }
                },
                _ => Err(reason.into()),
            },
        };
        resp
    };

    Ok(healthcheck.boxed())
}
```
The problem is that I can't convert the `azure_core::StatusCode` into `http::StatusCode` required for `HealthcheckError::Unknown`. On the line `StatusCode::from_u16(status.into())` I get an error that it can't be converted to u16.
I'm still new to Rust and I'm surprised that this doesn't work as it works with `http_types::StatusCode`